src/web_interface/components/database_routes.py

```diff
@@ -189,23 +207,24 @@ def _app_show_binary_search_results(self):
                 error = result
             elif result is not None:
                 yara_rules = make_unicode_string(yara_rules[0])
-                firmware_dict = self._build_firmware_dict_for_binary_search(result)
+                joined_results = self._join_results(result)
+                query_uid = self._do_binary_search_query(joined_results, yara_rules)
+                return redirect(url_for('database/browse', query=query_uid, only_firmwares=request.args.get('only_firmware')))
         else:
             error = 'No request ID found'
             request_id = None
         return render_template('database/database_binary_search_results.html', result=firmware_dict, error=error,
                                request_id=request_id, yara_rules=yara_rules)
 
-    def _build_firmware_dict_for_binary_search(self, uid_dict):
-        firmware_dict = {}
-        for rule in uid_dict:
-            with ConnectTo(FrontEndDbInterface, self._config) as connection:
-                firmware_list = [
-                    connection.firmwares.find_one(uid) or connection.file_objects.find_one(uid)
-                    for uid in uid_dict[rule]
-                ]
-                firmware_dict[rule] = sorted(connection.get_meta_list(firmware_list))
-        return firmware_dict
+    def _do_binary_search_query(self, binary_search_results: list, yara_rules: str) -> str:
```

Comment:
we are not doing/executing any searches here, right? So the name should probably be something like "save/store binary search query"?